### PR TITLE
ref(replay): Refactor replay details Timestamp onClick so activeTab is only updated when needed

### PR DESCRIPTION
**Pull request by ryan953**
So the point of all this is to move this call out from the hook, and only have it in the places that need it:

`setActiveTab(getFrameDetails(frame).tabKey);` is called from in the Timeline, UrlWalker and Breadcrumbs. No longer within the main tabs themselves.
We need this fix for the new Trace/Perf tab, because some breadcrumb types will be rendered here, in addition to the existing tabs/places (ie: clicks in Dom and Perf). Going forward `getFrameDetails()` will return the _preferred_ tab to show, we're not going to switch away from the current tab if that's what you're clicking on.

Along the way I made everything much more consistent around `useCrumbHandlers`.
- We don't add extra `useCallback` wrappers anymore, because we're going to re-render children based on `currentTime` so extra useCallbacks just mean extra refs for react to track
- We pass in `onMouse*` and `onClickTimestamp` consistently from the lists to the rows, some prop drilling is ok.
- Before we would sometimes leverage the `(frame: ReplayFrame) => void` signature, and sometimes just pass in `() => void`, now it's consistently the former.
- Now whenever we call `useCrumbHandlers` we're using all the return values. Before we were inconsistent, using some at one layer, and then more down below.

### CODE REVIEW

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx [hunk 1]
```diff
@@ -40,7 +43,10 @@ function ReplayTimelineEvents({
         <EventColumn key={column} column={column}>
           <Event
             frames={colFrames}
+            handleMouseEnter={handleMouseEnter}
+            handleMouseLeave={handleMouseLeave}
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:47]: Should these props be `onMouse*`?

File: static/app/utils/replays/hooks/useCrumbHandlers.tsx [hunk 2]
```diff
@@ -70,18 +69,15 @@ function useCrumbHandlers(startTimestampMs: number = 0) {
...
+  const onClickTimestamp = useCallback(
+    (frame: ReplayFrame) => setCurrentTime(frame.offsetMs),
+    [setCurrentTime]
   );
 
   return {
     handleMouseEnter,
     handleMouseLeave,
-    handleClick,
+    onClickTimestamp,
```
Comment (billyvg) [inline static/app/utils/replays/hooks/useCrumbHandlers.tsx:80]: The inconsistency here (and in `splitCrumb) with on vs handle is a bit confusing.

**Review by billyvg — COMMENTED** [review]
The handleMouse* vs onClick inconsistency is making it a bit confusing for me. Otherwise this looks good.

Did we do any testing in regards to perf and re-renders?

**Comment by billyvg** [comment]
re-rendering based on current time means we'll be re-rendering all rows every second?

File: static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/replays/breadcrumbs/replayTimelineEvents.tsx:47]: makes sense

https://develop.sentry.dev/frontend/#event-handlers

**Comment by ryan953** [comment]
> re-rendering based on current time means we'll be re-rendering all rows every second?

yes. but we were doing this before. the `render()` method runs whenever currentTime or currentHoverTime is updated because we might get the purple bar added. 

I'm testing and somehow the feedback button in the header is also re-rendering and causing paint flashes: breadcrumb and console rows do not trigger paints which is nice. 

I think we can try to extract the timestamp values from the rest of the replay context, to mitigate this issue. but it'll be a separate pr. 
All the `memo()` calls we had before, i don't think they were helping because we pass in `currentTime` as a prop in most cases, so things will re-render anyway.

**Review by billyvg — APPROVED** [review]

**Pull request merged**